test/ws.e2e-spec.ts

```diff
@@ -0,0 +1,646 @@
+        id: '1.0.1',
+        poolId: CONTRACT_ADDRESS,
+        from: 'A',
+        to: 'B',
+        amount: '5',
+        operator: 'A',
+        data: 'test',
+        timestamp: '2020-01-01 00:00:00Z',
+        rawOutput: {
+          from: 'A',
+          to: 'B',
+          operator: 'A',
+          value: '5',
+        },
+        transaction: {
+          blockNumber: '1',
+          transactionIndex: '0x0',
+          transactionHash: '0x123',
+          logIndex: '1',
+          signature: transferEventSignature,
+        },
+        type: 'fungible',
+      } as TokenTransferEvent,
+    };
+
+    await server
+      .ws('/api/ws')
+      .exec(() => {
+        expect(eventHandler).toBeDefined();
+        eventHandler([mockTransferEvent]);
+      })
+      .expectJson(message => {
+        expect(message.id).toBeDefined();
+        delete message.id;
+        expect(message).toEqual(mockTransferWebSocketMessage);
+        return true;
+      });
+  });
+
+  it('Websocket: token burn event', async () => {
+    eventstream.getSubscription.mockReturnValueOnce(<EventStreamSubscription>{
+      name: TOPIC + ':' + CONTRACT_ADDRESS,
+    });
+
+    http.get = jest.fn(
+      () =>
+        new FakeObservable(<EthConnectReturn>{
+          output: 'firefly://token/{id}',
+        }),
+    );
+
+    const mockBurnEvent: TransferEvent = {
+      subId: 'sb-123',
+      signature: transferEventSignature,
+      address: '',
+      blockNumber: '1',
+      transactionIndex: '0x0',
+      transactionHash: '0x123',
+      logIndex: '1',
+      timestamp: '2020-01-01 00:00:00Z',
+      data: {
+        from: 'B',
+        to: ZERO_ADDRESS,
+        operator: 'A',
+        value: '5',
+      },
+      inputMethod: 'burnWithData',
+      inputArgs: {
+        amount: '5',
+        data: '0x74657374',
+        from: 'B',
+      },
+    };
+
+    const mockBurnWebSocketMessage: WebSocketMessage = {
+      event: 'token-burn',
+      data: {
+        id: '1.0.1',
+        poolId: CONTRACT_ADDRESS,
+        from: 'B',
+        amount: '5',
+        operator: 'A',
+        data: 'test',
+        timestamp: '2020-01-01 00:00:00Z',
+        rawOutput: {
+          from: 'B',
+          to: ZERO_ADDRESS,
+          operator: 'A',
+          value: '5',
+        },
+        transaction: {
+          blockNumber: '1',
+          transactionIndex: '0x0',
+          transactionHash: '0x123',
+          logIndex: '1',
+          signature: transferEventSignature,
+        },
+        type: 'fungible',
+      } as TokenBurnEvent,
+    };
+
+    await server
+      .ws('/api/ws')
+      .exec(() => {
+        expect(eventHandler).toBeDefined();
+        eventHandler([mockBurnEvent]);
+      })
+      .expectJson(message => {
+        expect(message.id).toBeDefined();
+        delete message.id;
+        expect(message).toEqual(mockBurnWebSocketMessage);
+        return true;
+      });
+  });
+
+  it('Websocket: token transfer event from wrong pool', () => {
```

Comment:
So... this test was to test application-level event filtering on transfer subscriptions in ERC1155. For instance, if a subscription was to pool "N1" and received an event from pool "F1", it would ignore that event. You don't have application-level filtering of transfers since they are all locked to a specific contract instance...so this test isn't really needed.

However, I did suggest adding filtering of pool creation events. If you add that, an E2E test to verify that pool-specific subscriptions ignore the creation events from other pools would be good.